core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/MetricsServiceImpl.java

```diff
@@ -142,12 +142,18 @@ public int hashCode() {
     private DataAccess dataAccess;
 
     public void startUp(Session session, String keyspace, boolean resetDb) {
+        startUp(session, keyspace, resetDb, true);
+    }
+
+    public void startUp(Session session, String keyspace, boolean resetDb, boolean createSchema) {
         SchemaManager schemaManager = new SchemaManager(session);
         if (resetDb) {
             schemaManager.dropKeyspace(keyspace);
         }
-        // This creates/updates the keyspace + tables if needed
-        schemaManager.createSchema(keyspace);
+        if (createSchema) {
+            // This creates/updates the keyspace + tables if needed
```

Comment:
Mutliple session startups were somehow stepping on each other.  I'd be happy actually if this was re-written in a way that any client calling startUp could create the schema if needed.  Or even better, the schema should not be created by a client startup but perhaps an actual installer.  Anyway, this solved my startup issue and so I need it.  I do encourage you guys to make it better.